core/resharding.go

```diff
@@ -256,5 +271,43 @@ func GetInitShardState() types.ShardState {
 		}
 		shardState = append(shardState, com)
 	}
+	utils.GetLogInstance().Info("GetInitShardState", "shardState", shardState)
+	return shardState
+}
+
+// GetSwapOneShardState returns the initial shard state at genesis.
+func GetSwapOneShardState() types.ShardState {
```

Comment:
As per our discussion,  we will have a single function, which read shard state at end of each epoch. And it should read from db using key=epoch.